src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadataBuilder.cs

```diff
@@ -54,7 +54,9 @@ public class ClonedPackageSearchMetadata : IPackageSearchMetadata
             public async Task<PackageDeprecationMetadata> GetDeprecationMetadataAsync() => await (LazyDeprecationFactory ?? LazyNullDeprecationMetadata);
             public IEnumerable<PackageVulnerabilityMetadata> Vulnerabilities { get; set; }
             public bool IsListed { get; set; }
+            [Obsolete("PackagePath is recommended in place of PackageReader")]
             public Func<PackageReaderBase> PackageReader { get; set; }
+            public string PackagePath { get; set; }
```

Comment:
From a NuGet SDK perspective, I'm not sure I agree with the obsolete attribute. The advantage of `PackageReder`, even it should be a method and not a property, is that it automatically handles the feed type (v2 or v3 layout). From your change here, I'm assuming that customers wishing to avoid this obsolete property, they would have to reproduce this:
https://github.com/NuGet/NuGet.Client/pull/3823/files#diff-3e1afc29c5e3adc9ae44ffab25cca7b01b5bd898f43dd79a6f15dbba0dc14501R96-R105

Perhaps it would be better to add a `GetPackageReaderAsync(PackageIdentity package)` to `FindLocalPackagesResource` instead. That way the V2 and V3 resource implementations internalize this difference, keeping a single, simple API for callers to use.

Reply:
@zivkan and i talked through this feedback. Since it is a memberon ClonedPackageSearchMetadata, an innertype of PackageSearchMetadataBuilder, Andy is no longer worried about this approach.